### PR TITLE
feat: Use cookies instead of localstorage 

### LINKED ISSUE #3866: [ENG-267] Use cookies instead of localstorage for access and refresh token
Make sure to persist cookies even the browser is closed

<sub>[ENG-267](https://linear.app/heyxyz/issue/ENG-267)</sub>

### PR DESCRIPTION
**Pull request by Nirali-Ahir**
### <samp>🤖 Generated by Copilot at fa3a504</samp>

This pull request replaces the usage of local storage with cookies for storing user data and preferences in the web app. It adds the `js-cookie` library and its TypeScript types to the `package.json` file and updates the import and usage of the `CookieData` enum from the `@hey/data/storage` module in various files. It also removes the unused `Localstorage` enum and the `localStorage` object from the code. The main purpose of this change is to enable using cookies instead of local storage for storing user data and preferences.

/claim #3866

## Related issues

Fixes #3866

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at fa3a504</samp>

*  Add `js-cookie` and `@types/js-cookie` dependencies to `apps/web/package.json` to use the `js-cookie` library for managing cookies ([link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8R59), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-14b60f636e1a2b0061da57aaf231cb1ed15a5dc0c592425ed82e58fec95d42d8R93))
* Replace the import and use of `Localstorage` enum from `@hey/data/storage` with `CookieData` enum in various files to access the cookie keys instead of the local storage keys ([link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-329eeb50d1b4ae75b72b04d209e49f7ccffa8d57821c10fc5210d8d6c426f0d9L1-R1), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-fb52ca1ef04a77fdbe4aa1817146aff5b6344603331094d95e719c9a535c1f70L9-R13), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-5388758711a26ddb26630c668d4791adc2ef36f1c29adfd0e066685a0ec1f860L4-R4), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-53f4ad8d0d0cd0249226620ffd5b5c0e95cf9f9a3e4077efb08fba46c375fc57L4-R9), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-d295d024e33528411608bd1f7b2808696c12fee1985dadeb26441ffe7b7e0217L1-R9), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-4eee891e3d582a91a84683fcfd046ae5f34dbe6e989ed022c1bc0502f59b4c8eL4-R9), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-3c698a9b84ece6454b5795cf231a76fc8dd69531dd4ead68212460335f8a325bL3-R7), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-2d9af24d277e24d0e5b89b5e09bf7745dd1a4c55b9630b816484d0319313cacbL3-R8), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-29f5f49801e061d7d361d09c51d906806f8a75493cf5c6894c54f93d7dd624e5L5), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-5835d44291f23f60b40f852c5fa340e51e84746889167cefa1aa35e72f07213cL7-R7), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-e352bde6c4704d4fa7433e0c58b94c1386c56e3ba28b3dbceda7a6dfdd83af54L4-R4), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-522e9b1d3b32a7e320eedad337be78754c4058ccf8dc87a8969300c480e12ec8L4-R4), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-aa3d0cc0aedeb7153d71907759367332259538581bfaab8270434424f49d5d2eL3-R3), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-5b0b4ea0e7d567bdf2d7f676472626de6359c848ed3b8566fe3c853acdad39d0L2-R4), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-297bec618c22d7d0aeb2bc017c793b9a98e6e52816584e610886208450762baeL2-R4), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-11cb8d90f72090b1f16a9b7dd84f9651f929d363cec4daa88536df4185e58328L1-R11), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-c701aa17ecd80ac28459113f90db52363d7d23ae8f78e22be3d844ed47a0aae5L2-R3), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-9390cfccb3090a6b6b18dfaea3caaee07c9b76c4778a9ab1ac08ee61b8ca0c70L1-R1), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-0fdf6996e0be6ec6a1b52142c8dbc3d78da73f4072f92440ed764a75961ebd0aL1-R1))
* Add the import and use of `Cookies` object from `js-cookie` in various files to access and manipulate the cookie storage ([link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-5388758711a26ddb26630c668d4791adc2ef36f1c29adfd0e066685a0ec1f860R11), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-29f5f49801e061d7d361d09c51d906806f8a75493cf5c6894c54f93d7dd624e5R17), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-5835d44291f23f60b40f852c5fa340e51e84746889167cefa1aa35e72f07213cR13), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-e352bde6c4704d4fa7433e0c58b94c1386c56e3ba28b3dbceda7a6dfdd83af54R10), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-522e9b1d3b32a7e320eedad337be78754c4058ccf8dc87a8969300c480e12ec8R10), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-aa3d0cc0aedeb7153d71907759367332259538581bfaab8270434424f49d5d2eR9), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-c701aa17ecd80ac28459113f90db52363d7d23ae8f78e22be3d844ed47a0aae5L16-R17))
* Replace the use of `localStorage.getItem`, `localStorage.setItem`, and `localStorage.removeItem` with `Cookies.get`, `Cookies.set`, and `Cookies.remove` in various files to get, set, and remove data from the cookie storage ([link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-329eeb50d1b4ae75b72b04d209e49f7ccffa8d57821c10fc5210d8d6c426f0d9L14-R14), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-fb52ca1ef04a77fdbe4aa1817146aff5b6344603331094d95e719c9a535c1f70L43-R44), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-5388758711a26ddb26630c668d4791adc2ef36f1c29adfd0e066685a0ec1f860L63-R64), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-53f4ad8d0d0cd0249226620ffd5b5c0e95cf9f9a3e4077efb08fba46c375fc57L30-R31), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-53f4ad8d0d0cd0249226620ffd5b5c0e95cf9f9a3e4077efb08fba46c375fc57L67-R68), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-53f4ad8d0d0cd0249226620ffd5b5c0e95cf9f9a3e4077efb08fba46c375fc57L80-R81), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-4eee891e3d582a91a84683fcfd046ae5f34dbe6e989ed022c1bc0502f59b4c8eL33-R34), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-3c698a9b84ece6454b5795cf231a76fc8dd69531dd4ead68212460335f8a325bL28-R28), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-2d9af24d277e24d0e5b89b5e09bf7745dd1a4c55b9630b816484d0319313cacbL27-R28), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-29f5f49801e061d7d361d09c51d906806f8a75493cf5c6894c54f93d7dd624e5L114-R115), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-5835d44291f23f60b40f852c5fa340e51e84746889167cefa1aa35e72f07213cL44-R45), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-e352bde6c4704d4fa7433e0c58b94c1386c56e3ba28b3dbceda7a6dfdd83af54L34-R35), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-522e9b1d3b32a7e320eedad337be78754c4058ccf8dc87a8969300c480e12ec8L34-R35), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-aa3d0cc0aedeb7153d71907759367332259538581bfaab8270434424f49d5d2eL85-R86), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-5b0b4ea0e7d567bdf2d7f676472626de6359c848ed3b8566fe3c853acdad39d0L29-R30), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-297bec618c22d7d0aeb2bc017c793b9a98e6e52816584e610886208450762baeL110-R113), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-9390cfccb3090a6b6b18dfaea3caaee07c9b76c4778a9ab1ac08ee61b8ca0c70L40-R40), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-0fdf6996e0be6ec6a1b52142c8dbc3d78da73f4072f92440ed764a75961ebd0aL49-R49), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-0fdf6996e0be6ec6a1b52142c8dbc3d78da73f4072f92440ed764a75961ebd0aL87-R87))
* Replace the use of `buildLocalStorageKey` with `buildCookiesgeKey` in `useXmtpClient.tsx` to build the cookie key for storing the XMTP keys ([link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-297bec618c22d7d0aeb2bc017c793b9a98e6e52816584e610886208450762baeL16-R21), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-297bec618c22d7d0aeb2bc017c793b9a98e6e52816584e610886208450762baeL27-R32), [link](https://github.com/heyxyz/hey/pull/3885/files?diff=unified&w=0#diff-297bec618c22d7d0aeb2bc017c793b9a98e6e52816584e610886208450762baeL40-R41))

## Emoji

🍪🔑🌐

### CODE REVIEW

File: packages/data/storage.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
-// Localstorage keys
-export const Localstorage = {
+// Cookie item keys
+export const CookieData = {
```
Comment (bigint) [inline packages/data/storage.ts:2]: Just name it Cookie

**Review by bigint — CHANGES_REQUESTED** [review]

**Comment by bigint** [comment]
https://github.com/heyxyz/hey/assets/69431456/9f4764be-cd86-4e88-9101-4f10a9400528

it's not working for me 🙈

**Comment by Nirali-Ahir** [comment]
> Screen.Recording.2023-10-05.at.8.57.41.PM.mov 
> it's not working for me 🙈

can you help me to get access lens handle  ?

**Comment by Nirali-Ahir** [comment]
closing this in favour of #3889 --- now cookies only required for for access and refresh token so created new PR

**Comment by Nirali-Ahir** [comment]
closing this in favour of https://github.com/heyxyz/hey/pull/3889 --- now cookies only required for for access and refresh token so created new PR

**Pull request closed without merging**